pymc/pytensorf.py

```diff
@@ -177,6 +177,12 @@ def extract_obs_data(x: TensorVariable) -> np.ndarray:
             mask[mask_idx] = 1
             return np.ma.MaskedArray(array_data, mask)
 
+    from pymc.data import has_random_ancestor
+
+    if not has_random_ancestor(x):
+        cheap_eval_mode = Mode(linker="py", optimizer=None)
+        return x.eval(mode=cheap_eval_mode)
```

Comment:
Perhaps also check that `pm.pytensorf.inputvars(x)` is empty. A counter-example would be an explicit input in the graph, say `pt.scalar("x")`, that will fail with `MissingInput` when you try this eval.

Reply:
Good sugguestion. I've added that and a test for it

Reply:
Can you check how much time the creation of the cheap eval mode takes? Wondering if we should define it only once outside or if it's cheap enough that it doesn't matter